### PR TITLE
Сheck for path existence added [issue #6456]

### PR DESCRIPTION
**Pull request by vladimir-shcherbakov**
## Description

Issue #6456 fix.
## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [ ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/ResourceManager/Profile/Commands.Profile/AutoSave/EnableAzureRmContextAutosave.cs [hunk 1]
```diff
@@ -52,19 +52,22 @@ public override void ExecuteCmdlet()
             }
         }
 
-        void EnableAutosave(IAzureSession session, bool writeAutoSaveFile, out ContextAutosaveSettings result)
+        private static void EnableAutosave(IAzureSession session, bool writeAutoSaveFile, out ContextAutosaveSettings result)
```
Comment (markcowl) [inline src/ResourceManager/Profile/Commands.Profile/AutoSave/EnableAzureRmContextAutosave.cs]: I am not sure we want to make these methods static - in general, we want methods used in cmdlet execution to be non-static methods.

File: src/ResourceManager/Profile/Commands.Profile/AutoSave/EnableAzureRmContextAutosave.cs [hunk 2]
```diff
@@ -123,18 +124,19 @@ void EnableAutosave(IAzureSession session, bool writeAutoSaveFile, out ContextAu
...
-                valid = new FileInfo(path);
+                valid = new FileInfo(filepath);
             }
             catch
             {
                 // swallow any exception
             }
-            return valid != null;
+
+            return valid != null && Directory.Exists(valid.DirectoryName);
```
Comment (markcowl) [inline src/ResourceManager/Profile/Commands.Profile/AutoSave/EnableAzureRmContextAutosave.cs]: Should use FileUtilities to check existence here as we use it to ensure the directory is created above.

File: src/ResourceManager/Profile/Commands.Profile/AutoSave/EnableAzureRmContextAutosave.cs [hunk 3]
```diff
@@ -123,17 +124,19 @@ void EnableAutosave(IAzureSession session, bool writeAutoSaveFile, out ContextAu
...
 
-        bool IsValidPath(string path)
+        private bool IsValidPath(string filepath)
         {
             FileInfo valid = null;
             try
             {
-                valid = new FileInfo(path);
+                valid = new FileInfo(filepath);
+                FileUtilities.EnsureDirectoryExists(filepath);
```
[the remark this answers is not in the document]
Comment (vladimir-shcherbakov) [reply src/ResourceManager/Profile/Commands.Profile/AutoSave/EnableAzureRmContextAutosave.cs]: What needs to be verified?
* existence of the file
* existence of the containing directory (file may not exist at the moment)

if had the ci-build [test fail](https://travis-ci.org/Azure/azure-powershell/builds/406029707#L1511) on travis when I checked file existence.

**Comment by vladimir-shcherbakov** [comment]
Closing as
```FileUtilities.EnsureDirectoryExists``` method [gets called](https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/Profile/Commands.Profile/AutoSave/EnableAzureRmContextAutosave.cs#L89) in the constructor to create the directory if it doesn't exist. So, ```IsValidPath(string path)``` shouldn't check for path existence.

**Pull request closed without merging**